eng/pipelines/libraries/run-test-job.yml

```diff
@@ -64,14 +72,71 @@ jobs:
           - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
         - ${{ if ne(parameters.liveRuntimeBuildConfig, '') }}:
           - ${{ format('{0}_{1}_product_build_{2}{3}_{4}_{5}', parameters.runtimeFlavor, parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveRuntimeBuildConfig) }}
+      # SuperPMI collection needs to run mcs.exe on the AzDO machine. Assume that's an x64 machine, and download an x64 product build if needed.
+      - ${{ if and(eq(parameters.SuperPmiCollect, true), ne(parameters.archType, 'x64')) }}:
+          - ${{ format('{0}_{1}_product_build_{2}{3}_{4}_{5}', 'coreclr', '', parameters.osGroup, parameters.osSubgroup, 'x64', parameters.liveRuntimeBuildConfig) }}
 
       variables:
+
         - librariesTestsArtifactName: ${{ format('libraries_test_assets_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
         - _archiveTestsParameter: /p:ArchiveTests=true
 
+        # Variables used for SuperPMI collection
+        - ${{ if eq(parameters.SuperPmiCollect, true) }}:
+          - name: buildConfigUpper
+            ${{ if eq(parameters.buildConfig, 'debug') }}:
+              value: 'Debug'
+            ${{ if eq(parameters.buildConfig, 'release') }}:
+              value: 'Release'
+            ${{ if eq(parameters.buildConfig, 'checked') }}:
```

Comment:
I still wonder why we need this here. We don't have a checked configuration in libraries. Also, please move this and other variables that are only needed by the superpmi YML file into the YML file, i.e. `PythonScript` and others.

Reply:
I was just being consistent with the definition in the coreclr files.

> Also, please move this and other variables that are only needed by the superpmi YML file into the YML file, i.e. PythonScript and others.

I don't believe that's possible: the superpmi YML file is a "steps" template and as such can't define variables. (These here are not actual YML variables definitions either, they are "variables" parameters to a template that later injects them in a "variables:" section)